torch/quantization/fx/_equalize.py

```diff
@@ -220,3 +222,215 @@ def node_supports_equalization(node: Node, modules) -> bool:
     elif node.op == 'call_function':
         return node.target == nn.functional.linear
     return False
+
+def get_weight_eq_obs(node: Node, model: GraphModule, modules: Dict[str, nn.Module]):
+    """ Gets the following weight equalization observer. There should always
+    exsist a weight equalization observer after an input equalization observer.
+
+    Returns the node containing the weight equalization observer, and the weight
+    equalization observer if it has been newly created
+    """
+
+    # Find the next node that is either a nn.Linear or a functional layer
+    while (node.op not in ('output', 'call_function') and not
+           (node.op == 'call_module' and isinstance(node.target, str) and
+            isinstance(modules[node.target], nn.Linear))):
+        node = node.next
+
+    if node.op == 'call_module':
+        # If the next node is a nn.Linear layer, then it must have a
+        # WeightEqualizationObserver configuration
+        equalization_qconfig_map: Dict[str, Any] = model._equalization_qconfig_map  # type: ignore[assignment]
+        assert(equalization_qconfig_map.get(node.name, None) is not None)
+
+        weight_eq_obs = equalization_qconfig_map.get(node.name, None).weight()
```

Comment:
this creates a new instance of a weight observer.  Is this expected, or are we looking for the observer we inserted during the prepare step instead?

Reply:
Based on my understanding of the code, I thought for nn.Linear layers that the weight observers do not get inserted in the prepare step, and instead get created and run in [from_float](https://github.com/pytorch/pytorch/blob/master/torch/nn/quantized/modules/linear.py#L244) so I thought we would also need to create the weight equalization observer here. But if that isn't the case then I will fix the code!